File: spring-graal-native-image-samples/webflux-netty/src/main/java/com/example/demo/DemoApplication.java
package com.example.demo;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication(proxyBeanMethods = false)
public class DemoApplication {

	public static void main(String[] args) {
		SpringApplication.run(DemoApplication.class, args);
	}

	@RestController
	class Foo {

		@GetMapping("/")
		public String greet() {
			return "hi!";
		}
	}

}

File: spring-graal-native-samples/kotlin-webmvc/src/main/kotlin/com/example/demo/DemoApplication.kt
package com.example.demo

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication(proxyBeanMethods = false)
class KotlinCoroutinesApplication

fun main(args: Array<String>) {
	runApplication<KotlinCoroutinesApplication>(*args)
}

data class Bar(val message: String)

@RestController
class Foo {

	@GetMapping("/")
	fun greet(): Bar {
		return Bar("hi!")
	}
}
